### PR TITLE
Write unit tests for topic.js

### LINKED ISSUE #2314: Write unit tests for Topic.js
This issue concerns a server library that needs unit tests written for it.  Once #2309 lands, we'll have the capacity to write fast unit tests for this library that ensure the behavior stays the same.

The tests described here should be added to `tests/server-unit/*`.  You can run them with `npm run test:server-unit` to make sure that all tests pass.

### Topic.js
The following test cases should be covered in the unit tests for [Topic](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/lib/topic.js):
 - [ ] Test that topic's `.subscribe()` will respond to events `.publish()`ed on a channel.
 - [ ] Test that topics's `.publish()` will only trigger events listening to the published channel.
 - [ ] Test that topic's `unsubscribe()` will remove listeners from a channel.
 - [ ] Test that multiple calls to publish will trigger multiple calls to subscribed functions.
 - [ ] Test that `channels`, `events`, and `entities` are non empty.

Any other tests that seem appropriate can be added as needed.

### PR DESCRIPTION
**Pull request by jniles**
This PR adds tests for the topic.js library, a pub-sub system based on ioredis.

First, I have added a global setup function and teardown function (which is empty currently) for the server-unit tests.  These functions pre-load the environmental variables and setup up Chai with spies and promises.  These are located in `setup.js` in the `test/server-unit` directory.  See https://github.com/mochajs/mocha/issues/1460#issuecomment-93862610 for the suggestion for this code.

Second, I've added unit tests to the topic library.  They use a spy plugin for chai to detect when callbacks have been fired.  Unfortunately, due to bugs with the `chai-spies` library (https://github.com/chaijs/chai-spies/issues/78) I've had to use [chai-spies-next](https://www.npmjs.com/package/chai-spies-next).  In order to wait for the asynchronous code to complete, each spied function returns a
promise that determines when the testing should proceed.  If no actionhappens withing mocha's default timeout, the test fails.

Closes #2314.

### CODE REVIEW

File: test/server-unit/db.spec.js [hunk 1]
```diff
@@ -1,44 +1,34 @@
...
+  before(() => {
+    db = require('../../server/lib/db');
+  });
 
-const dbTest = function () {
 
-  it('Should check the connection to mysql', function (done) {
-    var fx = db.pool.getConnection(function (err, result) {
+  it('should check the connection to mysql', (done) => {
+    db.pool.getConnection(err => {
```
Comment (jeremielodi) [inline test/server-unit/db.spec.js:12]: cool, this is more readable

**Review by jeremielodi — COMMENTED** [review]
This test looks perfect, good work @jniles!, I should run it locally and see how it works

**Review by jeremielodi — COMMENTED** [review]
@jeremielodi approved this PR, it can be merged

**Comment by jniles** [comment]
Nice!  Try commenting "bors r+" and see if the robot will test and merge this PR for us.

**Pull request merged**